src/ws/Socket.ts

```diff
@@ -461,16 +462,17 @@ export class Socket extends EventEmitter {
 
         // If we already authed, it means we're reconnecting and should
         // establish authentication again.
-        if (this._authpacket) {
+        if (this._authpacket || this._optOutEventsArgs.length) {
             // tslint:disable-next-line no-floating-promises
-            this.call(authMethod, this._authpacket, { force: true })
+            const promise = this._optOutEventsArgs.length
```

Comment:
nit: inconsistent spacing- should be on new line like the other catches

Reply:
that's what prettier likes!